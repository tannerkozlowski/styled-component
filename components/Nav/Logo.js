import styled from 'styled-components'

import rem from '../../utils/rem'

const Logo = styled.div`
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  width: ${rem(164)};
  height: ${rem(40)};

  background-image: url(/static/nav-logo.png);
  background-position: center;
  background-size: contain;
`

export default Logo
