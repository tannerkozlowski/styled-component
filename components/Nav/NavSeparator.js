import styled from 'styled-components'

import rem from '../../utils/rem'

const NavSeparator = styled.span`
  width: ${rem(5)};
  height: ${rem(5)};
  margin: 0 ${rem(15)};

  border-radius: 50%;
  background: rgba(255, 255, 255, 0.35);
`

export default NavSeparator
