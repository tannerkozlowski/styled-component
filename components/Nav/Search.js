import React from 'react'
import styled, { css, createGlobalStyle } from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import PropTypes from 'prop-types'
import {
  lightGrey,
  grey,
  lightVioletRed,
  gold,
  darkGrey,
} from '../../utils/colors'
import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import { resetInput } from '../../utils/form'
import { mobile } from '../../utils/media'

const INPUT_ID = 'docs-search-input'

const Wrapper = styled.form`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex: 0 0 auto;
  ${mobile(css`
    display: block;
    padding: ${rem(10)};
    span.algolia-autocomplete {
      display: block !important;
    }
    span.ds-dropdown-menu::before {
      display: none;
    }
  `)};
`

const Input = styled.input`
  ${resetInput};
  flex: 0 0 auto;
  width: ${rem(130)};
  line-height: ${rem(navbarHeight - 20)};
  font-size: ${rem(15)};
  color: currentColor;
  ::placeholder {
    color: currentColor;
    opacity: 0.5;
  }
  &:focus {
    border-bottom: 1px solid currentColor;
  }
  ${mobile(css`
    padding-left: ${rem(40)};
    display: block;
    width: 100%;
    border: 1px solid ${lightGrey};
    border-radius: ${rem(3)};
    background: ${grey};
    color: white;
    &:focus {
      border-bottom: 1px solid ${lightGrey};
    }
  `)};
`

const StyledSearchIcon = styled(FontAwesomeIcon).attrs({
  icon: faSearch,
})``

const Button = styled.label.attrs({
  for: INPUT_ID,
})`
  ${resetInput};
  flex: 0 0 auto;
  ${resetInput} flex: 0 0 auto;
  margin-right: ${rem(4)};
  cursor: pointer;
  display: flex;
  &:hover,
  &:focus {
    opacity: 0.7;
    filter: drop-shadow(0 1px 2px rgba(0, 0, 0, 0.2));
  }

  &:active {
    background: rgba(0, 0, 0, 0.1);
  }

  ${StyledSearchIcon} {
    width: ${rem(20)};
    height: ${rem(20)};

    path {
      fill: currentColor;
    }
  }
  ${mobile(css`
    position: absolute;
    height: ${rem(34)};
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${rem(34)};
    z-index: 1;
    background: ${grey};
    color: white;
    border-radius: ${rem(3)};
  `)};
`

const SearchButton = () => (
  <Button>
    <StyledSearchIcon />
  </Button>
)

const GlobalStyles = createGlobalStyle`
    .algolia-autocomplete
      .ds-dropdown-menu
      .ds-suggestion.ds-cursor
      .algolia-docsearch-suggestion.suggestion-layout-simple,
    .algolia-autocomplete
      .ds-dropdown-menu
      .ds-suggestion.ds-cursor
      .algolia-docsearch-suggestion:not(.suggestion-layout-simple)
      .algolia-docsearch-suggestion--content {
      background: ${lightVioletRed};
    }

    /* Main category (eg. Getting Started) */
    .algolia-autocomplete .algolia-docsearch-suggestion--category-header {
      color: ${darkGrey};
    }

    /* Category (eg. Downloads) */
    .algolia-autocomplete .algolia-docsearch-suggestion--subcategory-column {
      color: ${grey};
    }

    /* Title (eg. Bootstrap CDN) */
    .algolia-autocomplete .algolia-docsearch-suggestion--title {
      font-weight: bold;
      color: black;
    }

    /* Description description (eg. Bootstrap currently works...) */
    .algolia-autocomplete .algolia-docsearch-suggestion--text {
      color: ${grey};
    }

    /* Highlighted text */
    .algolia-autocomplete .algolia-docsearch-suggestion--highlight {
      color: ${gold};
      background: transparent;
    }
`

const Search = ({ isDocs, className }) => (
  <Wrapper className={className}>
    <GlobalStyles />
    <SearchButton />
    <Input
      id={INPUT_ID}
      placeholder={isDocs ? `Search ...` : `Search docs ...`}
    />
  </Wrapper>
)

Search.propTypes = {
  className: PropTypes.string,
}

export default Search
