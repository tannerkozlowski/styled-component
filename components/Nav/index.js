import React, { Component } from 'react'
import Sidebar from './Sidebar'
import SidebarMenu from './SidebarMenu'

class Nav extends Component {
  state = {
    isNavbarMenuFolded: true
  }

  render() {
    const { onRouteChange, isSideFolded } = this.props

    return (
      <div>
        <Sidebar isFolded={isSideFolded}>
          <SidebarMenu onRouteChange={onRouteChange} />
        </Sidebar>
      </div>
    )
  }
}

export default Nav
