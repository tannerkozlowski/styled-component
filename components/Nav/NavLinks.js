import React from 'react'
import styled from 'styled-components'

import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import NavSeparator from './NavSeparator'
import Link from '../Link'

const Wrapper = styled.nav`
  display: flex;
  align-items: center;
`

const NavLink = styled(Link).attrs({
  unstyled: true,
})`
  display: inline-block;
  line-height: ${rem(navbarHeight)};
  transition: opacity 200ms, transfrom 200ms;
  cursor: pointer;

  letter-spacing: ${rem(.4)}
  color: currentColor;

  &:hover,
  &:focus {
    opacity: 0.8;
  }

  &:active {
    transform: scale(0.95);
    opacity: 0.6;
  }
`

const NavLinks = () => (
  <Wrapper>
    <NavLink>Concepts</NavLink>
    <NavSeparator />
    <NavLink>Try it out</NavLink>
    <NavSeparator />
    <NavLink>Documentation</NavLink>
  </Wrapper>
)

export default NavLinks
