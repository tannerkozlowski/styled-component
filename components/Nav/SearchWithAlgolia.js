import React, { Component } from 'react'
import Router from 'next/router'
import PropTypes from 'prop-types'
import Search from './Search'

class SearchWithAlgolia extends Component {
  static propTypes = {
    className: PropTypes.string,
    requestModalClose: PropTypes.func.isRequired,
  }

  isDocs = process.browser && Router.pathname.startsWith('/docs')

  componentDidMount() {
    if (process.env.NODE_ENV !== 'test') {
      import('docsearch.js').then(mdl => {
        mdl.default({
          apiKey: '79886fb59ad3ebe2002b481cffbbe7cb',
          indexName: 'styled-components',
          inputSelector: '[class^="Search__Input"]',
          debug: true, // Set debug to true if you want to inspect the dropdown
          handleSelected: (input, event, suggestion) => {
            // original handleselect
            this.props.requestModalClose()
            input.setVal('')
            window.location.assign(suggestion.url)
          },
        })
      })
    }
  }

  render() {
    return <Search isDocs={this.isDocs} className={this.props.className} />
  }
}

export default SearchWithAlgolia
