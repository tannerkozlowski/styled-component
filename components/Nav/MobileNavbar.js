import React, { PureComponent } from 'react'
import styled, { css } from 'styled-components'

import rem from '../../utils/rem'
import { navbarHeight } from '../../utils/sizes'
import { paleGrey } from '../../utils/colors'
import { mobile } from '../../utils/media'
import Link from '../Link'
import NavLinks from './NavLinks'
import Social from './Social'
import Search from './Search'
import Logo from './Logo'
import NavSeparator from './NavSeparator'
import NavButton from './NavButton'

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: ${rem(navbarHeight)};
  `)}
`

const SecondaryMenu = styled.div`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p => p.open ? css`
    height: ${rem(navbarHeight)};
  ` : css`
    height: 0;
  `}

  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  -webkit-user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`

const LogoLink = styled(Link).attrs({
  unstyled: true,
  href: '/',
})`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`

const ArrowWrapper = styled.div`
  transition: transform 0.1s;

  ${p => p.rotate && css`
    transform-origin: 50% 55%;
    transform: rotate(180deg);
  `}
`

const PaddedSocial = styled(Social)`
  padding-right: ${rem(20)};
`

const CloseIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width="15" height="15" xmlnsXlink="http://www.w3.org/1999/xlink">
    <title>close</title>
    <use fill="#FFF" xlinkHref="#close" transform="translate(1 1)" />
    <defs>
      <path id="close" d="M-.7.7l13 13 1.4-1.4-13-13L-.7.7zm13-1.4l-13 13 1.4 1.4 13-13-1.4-1.4z"/>
    </defs>
  </svg>
)

const FoldIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" width="17" height="14" xmlnsXlink="http://www.w3.org/1999/xlink">
    <title>fold</title>
    <use fill="#FFF" xlinkHref="#fold" transform="translate(0 1)"/>
    <defs>
      <path id="fold" d="M0 1h17v-2H0v2zm17 4H0v2h17V5zM0 13h17v-2H0v2z"/>
    </defs>
  </svg>
)

const ArrowIcon = props => (
  <ArrowWrapper {...props}>
    <svg xmlns="http://www.w3.org/2000/svg" width="12" height="8" xmlnsXlink="http://www.w3.org/1999/xlink">
      <title>arrow down</title>
      <use fill="#FFF" xlinkHref="#menuArrow" transform="translate(1 1)"/>
      <defs>
        <path id="menuArrow" d="M5 5l-.7.7.7.7.7-.7L5 5zM9.3-.7l-5 5 1.4 1.4 5-5L9.3-.7zm-3.6 5l-5-5L-.7.7l5 5 1.4-1.4z"/>
      </defs>
    </svg>
  </ArrowWrapper>
)

class MobileNavbar extends PureComponent {
  state = {
    isMenuOpen: false,
  }

  render() {
    const { isMenuOpen } = this.state
    const { isFolded, onFold } = this.props

    return (
      <Wrapper>
        <NavButton
          active={!isFolded}
          onClick={onFold}
        >
          {isFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>

        <LogoLink>
          <Logo compact />
        </LogoLink>

        <NavButton
          onClick={this.toggleMenu}
          active={isMenuOpen}
        >
          <ArrowIcon rotate={isMenuOpen} />
        </NavButton>

        <SecondaryMenu open={isMenuOpen}>
          <NavLinks />
          <NavSeparator />
          <Search />
          <NavSeparator />
          <PaddedSocial />
        </SecondaryMenu>
      </Wrapper>
    )
  }

  toggleMenu = () => {
    this.setState({ isMenuOpen: !this.state.isMenuOpen })
  }
}

export default MobileNavbar
