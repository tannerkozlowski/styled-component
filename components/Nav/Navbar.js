import React, { PureComponent, createRef } from 'react'
import styled, { css } from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTimes } from '@fortawesome/free-solid-svg-icons'
import rem from '../../utils/rem'
import { violetRed, paleGrey, gold } from '../../utils/colors'
import { navbarHeight, searchModalWidth } from '../../utils/sizes'
import { headerFont } from '../../utils/fonts'
import { mobile } from '../../utils/media'
import Link from '../Link'
import NavLinks from './NavLinks'
import Social from './Social'
import Logo from './Logo'
import MobileNavbar from './MobileNavbar'
import SearchWithAlgolia from './SearchWithAlgolia'

const Wrapper = styled.nav`
  position: fixed;
  left: 0;
  box-sizing: border-box;
  z-index: 3;

  width: 100%;
  height: ${rem(navbarHeight)};

  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  background: ${props => (props.transparent ? 'transparent' : violetRed)};
  transition: background 300ms ease-out;
  color: white;
`

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`
/* stylelint-disable */
const StyledSocial = styled(Social)``
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;
  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`

const StyledModalCloseIcon = styled(FontAwesomeIcon).attrs({
  icon: faTimes,
})`
  width: ${rem(20)};
  height: ${rem(20)};
  color: ${violetRed};
`

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    box-shadow: 0 1px 6px -2px black;
    position: absolute;
    top: ${rem(20)};
    left: 50%;
    border-radius: ${rem(3)};
    transform: translate(-50%, 0);
    width: ${rem(searchModalWidth)};
  `)};
`

const AlgoliaModalHeader = styled.div`
  display: none;
  color: currentColor;
  ${mobile(css`
    display: block;
    button {
      position: absolute;
      left: 100%;
      top: 0;
      border: none;
      cursor: pointer;
      box-shadow: 0 1px 6px -2px black;
      transform: translate(-50%, -50%);
      border-radius: 50%;
      color: ${gold};
      display: flex;
      background: #fff;
    }
  `)};
`

const AlgoliaModalOverlay = styled.div`
  ${mobile(css`
    position: fixed;
    top: 0;
    z-index: 1;
    left: 0;
    display: ${props => (props.isOpen ? 'block' : 'none')};
    height: 100%;
    width: 100%;
    background: ${paleGrey};

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`

class ModalContainer extends PureComponent {
  modalElement = createRef()
  onModalOverlayClick = e => {
    if (!this.modalElement.current.contains(e.target)) {
      this.props.requestModalClose()
    }
  }
  onCloseButtonClick = e => {
    e.stopPropagation()
    this.props.requestModalClose()
  }
  render() {
    return (
      <AlgoliaModalOverlay
        onClick={this.onModalOverlayClick}
        isOpen={this.props.isOpen}
      >
        <AlgoliaModal ref={this.modalElement}>
          <AlgoliaModalHeader>
            <button onClick={this.onCloseButtonClick}>
              <StyledModalCloseIcon />
            </button>
          </AlgoliaModalHeader>
          <div>{this.props.children}</div>
        </AlgoliaModal>
      </AlgoliaModalOverlay>
    )
  }
}

const LogoLink = styled(Link).attrs({
  unstyled: true,
  href: '/',
})`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`

class Navbar extends PureComponent {
  state = {
    isOpen: false,
  }
  openModal = () => this.setState(() => ({ isOpen: true }))
  closeModal = () =>
    this.setState(({ isOpen }) => (isOpen ? { isOpen: false } : null))
  render() {
    const {
      onSideToggle,
      onMobileNavToggle,
      isSideFolded,
      isMobileNavFolded,
      showSideNav,
      transparent,
    } = this.props

    return (
      <Wrapper transparent={transparent}>
        <NormalNavbar>
          <StartWrapper>
            <LogoLink>
              <Logo />
            </LogoLink>
            <NavLinks />
          </StartWrapper>
          <EndWrapper>
            <ModalContainer
              isOpen={this.state.isOpen}
              requestModalClose={this.closeModal}
            >
              <SearchWithAlgolia requestModalClose={this.closeModal} />
            </ModalContainer>
            <StyledSocial />
          </EndWrapper>
        </NormalNavbar>
        <MobileNavbar
          isSideFolded={isSideFolded}
          onSearchButtonClick={this.openModal}
          isMobileNavFolded={isMobileNavFolded}
          onSideToggle={onSideToggle}
          onMobileNavToggle={onMobileNavToggle}
          showSideNav={showSideNav}
        />
      </Wrapper>
    )
  }
}

export default Navbar
