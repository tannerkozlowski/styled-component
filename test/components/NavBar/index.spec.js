import React from 'react'
import { mount } from 'enzyme'
import 'jest-styled-components'
import Nav from '../../../components/Nav'

test('Nav renders correctly', () => {
  const wrapper = mount(<Nav />)

  expect(wrapper).toMatchSnapshot()
})
