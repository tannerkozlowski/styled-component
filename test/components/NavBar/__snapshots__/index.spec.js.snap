// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
.c0 {
  position: fixed;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  display: block;
  z-index: 5;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  left: 0;
  top: 0;
  bottom: 0;
  right: auto;
  width: 16.666666666666668rem;
  height: 100%;
  background: #efefef;
  box-sizing: border-box;
  color: inherit;
  overflow-y: scroll;
  -webkit-transition: -webkit-transform 150ms ease-out;
  -webkit-transition: transform 150ms ease-out;
  transition: transform 150ms ease-out;
}

.c4 {
  display: inline-block;
  text-decoration: none;
  color: inherit;
  cursor: pointer;
  padding: 0.1111111111111111rem 0.4444444444444444rem;
  margin: -0.1111111111111111rem -0.4444444444444444rem;
}

.c1 {
  display: block;
  box-sizing: border-box;
  height: 100%;
  padding-top: 1.3888888888888888rem;
}

.c2 {
  margin-bottom: 1.6666666666666667rem;
}

.c3 {
  display: block;
  margin: 0.5555555555555556rem 2.2222222222222223rem;
  font-weight: normal;
}

.c5 {
  display: block;
  margin: 0.5555555555555556rem 2.2222222222222223rem 0.5555555555555556rem 3.0555555555555554rem;
  font-size: 0.9rem;
  font-weight: normal;
}

@media (min-width:62.5em) {
  .c4 {
    border-radius: 0.16666666666666666rem;
  }

  .c4:hover {
    background: rgba(20,20,20,0.1);
  }
}

<Nav>
  <div>
    <styled.nav>
      <styled.nav>
        <nav
          className="c0"
        >
          <SidebarMenu>
            <styled.div>
              <div
                className="c1"
              >
                <styled.div>
                  <div
                    className="c2"
                  >
                    <styled.h4>
                      <h4
                        className="c3"
                      >
                        <Link
                          href="/docs/basics"
                        >
                          <Link
                            href="/docs/basics"
                          >
                            <styled.a
                              href="/docs/basics"
                              onClick={[Function]}
                            >
                              <a
                                className="c4"
                                href="/docs/basics"
                                onClick={[Function]}
                              >
                                Basics
                              </a>
                            </styled.a>
                          </Link>
                        </Link>
                      </h4>
                    </styled.h4>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#motivation"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#motivation"
                          >
                            Motivation
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#installation"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#installation"
                          >
                            Installation
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#getting-started"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#getting-started"
                          >
                            Getting Started
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#passed-props"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#passed-props"
                          >
                            Passed props
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#adapting-based-on-props"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#adapting-based-on-props"
                          >
                            Adapting based on props
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#styling-any-components"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#styling-any-components"
                          >
                            Styling any components
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#extending-styles"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#extending-styles"
                          >
                            Extending styles
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#attaching-additional-props"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#attaching-additional-props"
                          >
                            Attaching additional props
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#animations"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#animations"
                          >
                            Animations
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/basics#react-native"
                        >
                          <a
                            className="c4"
                            href="/docs/basics#react-native"
                          >
                            React Native
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c2"
                  >
                    <styled.h4>
                      <h4
                        className="c3"
                      >
                        <Link
                          href="/docs/advanced"
                        >
                          <Link
                            href="/docs/advanced"
                          >
                            <styled.a
                              href="/docs/advanced"
                              onClick={[Function]}
                            >
                              <a
                                className="c4"
                                href="/docs/advanced"
                                onClick={[Function]}
                              >
                                Advanced
                              </a>
                            </styled.a>
                          </Link>
                        </Link>
                      </h4>
                    </styled.h4>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#theming"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#theming"
                          >
                            Theming
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#refs"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#refs"
                          >
                            Refs
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#security"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#security"
                          >
                            Security
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#existing-css"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#existing-css"
                          >
                            Existing CSS
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#media-templates"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#media-templates"
                          >
                            Media Templates
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#tagged-template-literals"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#tagged-template-literals"
                          >
                            Tagged Template Literals
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/advanced#server-side-rendering"
                        >
                          <a
                            className="c4"
                            href="/docs/advanced#server-side-rendering"
                          >
                            Server Side Rendering
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c2"
                  >
                    <styled.h4>
                      <h4
                        className="c3"
                      >
                        <Link
                          href="/docs/api"
                        >
                          <Link
                            href="/docs/api"
                          >
                            <styled.a
                              href="/docs/api"
                              onClick={[Function]}
                            >
                              <a
                                className="c4"
                                href="/docs/api"
                                onClick={[Function]}
                              >
                                API Reference
                              </a>
                            </styled.a>
                          </Link>
                        </Link>
                      </h4>
                    </styled.h4>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/api#primary"
                        >
                          <a
                            className="c4"
                            href="/docs/api#primary"
                          >
                            Primary
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/api#helpers"
                        >
                          <a
                            className="c4"
                            href="/docs/api#helpers"
                          >
                            Helpers
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/api#supported-css"
                        >
                          <a
                            className="c4"
                            href="/docs/api#supported-css"
                          >
                            Supported CSS
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/api#flow"
                        >
                          <a
                            className="c4"
                            href="/docs/api#flow"
                          >
                            Flow
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/api#typescript"
                        >
                          <a
                            className="c4"
                            href="/docs/api#typescript"
                          >
                            TypeScript
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c2"
                  >
                    <styled.h4>
                      <h4
                        className="c3"
                      >
                        <Link
                          href="/docs/tooling"
                        >
                          <Link
                            href="/docs/tooling"
                          >
                            <styled.a
                              href="/docs/tooling"
                              onClick={[Function]}
                            >
                              <a
                                className="c4"
                                href="/docs/tooling"
                                onClick={[Function]}
                              >
                                Tooling
                              </a>
                            </styled.a>
                          </Link>
                        </Link>
                      </h4>
                    </styled.h4>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/tooling#babel-plugin"
                        >
                          <a
                            className="c4"
                            href="/docs/tooling#babel-plugin"
                          >
                            Babel Plugin
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/tooling#test-utilities"
                        >
                          <a
                            className="c4"
                            href="/docs/tooling#test-utilities"
                          >
                            Test Utilities
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/tooling#stylelint"
                        >
                          <a
                            className="c4"
                            href="/docs/tooling#stylelint"
                          >
                            Stylelint
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c2"
                  >
                    <styled.h4>
                      <h4
                        className="c3"
                      >
                        <Link
                          href="/docs/faqs"
                        >
                          <Link
                            href="/docs/faqs"
                          >
                            <styled.a
                              href="/docs/faqs"
                              onClick={[Function]}
                            >
                              <a
                                className="c4"
                                href="/docs/faqs"
                                onClick={[Function]}
                              >
                                FAQs
                              </a>
                            </styled.a>
                          </Link>
                        </Link>
                      </h4>
                    </styled.h4>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/faqs#can-i-nest-rules"
                        >
                          <a
                            className="c4"
                            href="/docs/faqs#can-i-nest-rules"
                          >
                            Can I nest rules?
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/faqs#can-i-refer-to-other-components"
                        >
                          <a
                            className="c4"
                            href="/docs/faqs#can-i-refer-to-other-components"
                          >
                            Can I refer to other components?
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                    <styled.h5>
                      <h5
                        className="c5"
                      >
                        <styled.a
                          href="/docs/faqs#can-i-use-css-frameworks"
                        >
                          <a
                            className="c4"
                            href="/docs/faqs#can-i-use-css-frameworks"
                          >
                            Can I use css frameworks?
                          </a>
                        </styled.a>
                      </h5>
                    </styled.h5>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </SidebarMenu>
        </nav>
      </styled.nav>
    </styled.nav>
  </div>
</Nav>
`;
